test/e2e/member_operator_test.go

```diff
@@ -0,0 +1,115 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/common"
+	userv1 "github.com/openshift/api/user/v1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestUserAccount(t *testing.T) {
+	userAccList := &toolchainv1alpha1.UserAccountList{}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, userAccList)
+	require.NoError(t, err, "failed to add custom resource scheme to framework")
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	client := f.Client.Client
+
+	// wait for operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "member-operator", 1, retryInterval, timeout)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("member-operator is ready and running state")
+
+	// create useraccount
+	userAcc := newUserAcc(t, f, ctx)
+	err = f.Client.Create(context.TODO(), userAcc, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user account '%s' created", userAcc.Name)
+
+	t.Run("verify_useraccount_ok", func(t *testing.T) {
```

Comment:
Add a test for UserAccount deletion. To check if the corresponding secondary resources are deleted too.

Reply:
@alexeykazakov currently our controller returns if `UserAccount` not found [link](https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L91).  I guess, we first need to handle this into useraccount_ctrl then we can add test for the same :+1:

Reply:
The cascade deletion / garbage collection is handled by Kubernetes automatically if we set owner reference correctly. So it should already work.

Reply:
Delete of `UserAccount` is NOT deleting corresponding `User` and `Identity`.  We need to check this.

Reply:
Hm... It should delete them but I also can confirm it doesn't work. At least not in my minishift. See https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/
Even explicit `oc delete useraccount <...> --cascade=true` (which should be used by default) doesn't help. We need to investigate that.